ark/phenotyping/som_utils_test.py

```diff
@@ -835,39 +913,63 @@ def test_create_fov_pixel_data():
         assert np.all(sample_pixel_mat.loc[:, ['chan0', 'chan1']].sum(axis=1).values != 0)
 
         # assert the size of the subsetted DataFrame is 0.1 of the preprocessed DataFrame
-        assert sample_pixel_mat.shape[0] * 0.1 == sample_pixel_mat_subset.shape[0]
+        # NOTE: need to account for rounding if multiplying by 0.1 leads to non-int
+        assert round(sample_pixel_mat.shape[0] * 0.1) == sample_pixel_mat_subset.shape[0]
+
+        # TEST 3: run fov preprocessing with a custom pixel_norm_val (between 0 and 1)
+        # NOTE: divide by 1000 and set pixel_norm_val to 0.1475 to ensure rows get removed
+        sample_pixel_mat, sample_pixel_mat_subset = som_utils.create_fov_pixel_data(
+            fov=fov, channels=chans, img_data=sample_img_data / 1000, seg_labels=seg_labels,
+            pixel_norm_val=0.1475
+        )
+
+        # assert the channel names are the same
+        misc_utils.verify_same_elements(flowsom_chans=sample_pixel_mat.columns.values[:-4],
+                                        provided_chans=chans)
+        misc_utils.verify_same_elements(flowsom_chans=sample_pixel_mat_subset.columns.values[:-4],
+                                        provided_chans=chans)
+
+        # assert no rows sum to 0
+        assert np.all(sample_pixel_mat.loc[:, ['chan0', 'chan1']].sum(axis=1).values != 0)
```

Comment:
Is there a more precise check we could add here to ensure that this addition is working? This test would pass with the previous version and this version

Reply:
@ngreenwald because we're dividing by row sums, we can change this test to ensure that all rows sum to 1 to test normalization with different `pixel_norm_val` parameters. Is this what you were thinking about, or something more specific?

Reply:
I mean something that checks that passing `pixel_norm_val` is working as intended. For example, an expected decrease in the total amount of pixels included in the df or something like that

Reply:
@ngreenwald oh ok. I can do something along the lines of:

```
assert sample_pixel_mat.shape[0] < (sample_img_data.shape[0] * sample_img_data.shape[1])
```

This will assert that we actually generated fewer pixels in `sample_pixel_mat` than there exist in `sample_img_data`.

The opposite test:

```
assert sample_pixel_mat.shape[0] == (sample_img_data.shape[0] * sample_img_data.shape[1])
```

would be good for the other 2 tests where no pixels are removed by `pixel_norm_val`.